ChainGenerator/Components/Widgets/WidgetGenerator.razor

```diff
@@ -1,6 +1,12 @@
 ﻿@using System.Text
 @inject IServiceProvider ServiceProvider
 
+<!-- The badge icon which indicates an ongoing refresh -->
+@if (refreshInProgress)
+{<MudBadge Origin="Origin.TopLeft" Color="Color.Warning" Overlap="true" Bordered="false" Icon="@badgeIcon" ></MudBadge>}
```

Comment:
I recommend putting these on different lines. It just helps a little with the readability of the code. You can use lines 16-23 as a reference.